account_payment_partner/tests/test_account_payment_partner.py

```diff
@@ -0,0 +1,588 @@
+        # Test payment mode change on aml
+        mode = self.supplier_payment_mode.copy()
+        aml.payment_mode_id = mode
+        self.assertEqual(invoice.payment_mode_id, mode)
+        # Test payment mode editability on account move
+        self.assertFalse(invoice.has_reconciled_items)
+        invoice.payment_mode_id = self.supplier_payment_mode
+        self.assertEqual(aml.payment_mode_id, self.supplier_payment_mode)
+
+    def test_invoice_create_out_invoice(self):
+        invoice = self._create_invoice(
+            default_move_type="out_invoice", partner=self.customer
+        )
+        invoice.action_post()
+        aml = invoice.line_ids.filtered(
+            lambda x: x.account_id.account_type == "asset_receivable"
+        )
+        self.assertEqual(invoice.payment_mode_id, aml[0].payment_mode_id)
+
+    def test_invoice_create_out_refund(self):
+        self.manual_out.bank_account_required = False
+        invoice = self._create_invoice(
+            default_move_type="out_refund", partner=self.customer
+        )
+        invoice.action_post()
+        self.assertEqual(
+            invoice.payment_mode_id,
+            self.customer.customer_payment_mode_id.refund_payment_mode_id,
+        )
+
+    def test_invoice_create_in_refund(self):
+        self.manual_in.bank_account_required = False
+        invoice = self._create_invoice(
+            default_move_type="in_refund", partner=self.supplier
+        )
+        invoice.action_post()
+        self.assertEqual(
+            invoice.payment_mode_id,
+            self.supplier.supplier_payment_mode_id.refund_payment_mode_id,
+        )
+
+    def test_invoice_constrains(self):
+        with self.assertRaises(UserError):
+            self.move_model.create(
+                {
+                    "partner_id": self.supplier.id,
+                    "move_type": "in_invoice",
+                    "invoice_date": fields.Date.today(),
+                    "company_id": self.company.id,
+                    "payment_mode_id": self.supplier_payment_mode_c2.id,
+                }
+            )
+
+    def test_payment_mode_constrains_01(self):
+        self.move_model.create(
+            {
+                "partner_id": self.supplier.id,
+                "move_type": "in_invoice",
+                "invoice_date": fields.Date.today(),
+                "company_id": self.company.id,
+            }
+        )
+        with self.assertRaises(UserError):
+            self.supplier_payment_mode.company_id = self.company_2
+
+    def test_payment_mode_constrains_02(self):
+        self.move_model.create(
+            {
+                "date": fields.Date.today(),
+                "journal_id": self.journal_sale.id,
+                "name": "/",
+                "ref": "reference",
+                "state": "draft",
+                "invoice_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
```

Comment:
Could you please use [Command](https://github.com/odoo/odoo/blob/f45451b98e2e12aa803f1abc0ba3c9a6f90310f3/odoo/fields.py#L4227
), it's more readable?. And check the remaining code, please.
```suggestion
                    Command.create(
                        {
```